src/MyEventEmitter.js

```diff
@@ -1,14 +1,70 @@
 'use strict';
 
 class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  constructor() {
+    this.listeners = [];
+  }
+
+  on(eventName, listener) {
+    const listenerObj = { name: eventName, function: listener, once: false };
+
+    this.listeners.push(listenerObj);
+  }
+
+  once(eventName, listener) {
+    const listenerObj = { name: eventName, function: listener, once: true };
+
+    this.listeners.push(listenerObj);
+  }
+
+  off(eventName, listener) {
+    this.listeners = [...this.listeners].filter(
+      (obj) => !(obj.name === eventName && obj.function === listener),
+    );
+  }
+
+  emit(eventName, ...args) {
+    const targetListeners = [...this.listeners].filter(
+      (listener) => listener.name === eventName,
+    );
```

Comment:
Similar to the previous comment, creating a shallow copy of the listeners array is unnecessary here. You can filter directly on `this.listeners`.